pkg/controllers/dnsrecord/dnsrecord_controller.go

```diff
@@ -39,12 +40,13 @@ const (
 )
 
 var Clock clock.Clock = clock.RealClock{}
+var DNSProvider dns.Provider
```

Comment:
This looks to be a package level var that is set in the reconcile and used in other functions? This could lead to odd behaviour if multiple threads of execution  as they would change the value of this for all the functions using that var